schema.ts

```diff
@@ -1374,6 +1375,16 @@ export interface ObjectSchemaDefinition<K extends string, L extends string>
    */
   index?: IndexDefinition;
 
+  /**
+   * The name of the property within {@link ObjectSchemaDefinition.properties} that can be be interpreted as
+   * text/number representing the version of an item.
+   *
+   * Must be a {@link ValueType.String} or {@link ValueType.Number} property.
+   * TODO(ebo): Unhide this
+   * @hidden
+   */
+  itemVersionProperty? : PropertyIdentifier<K>;
```

Comment:
Naming suggestion: `versionProperty`.

The term `item` is redundant, as all of these refer to the item (it's `titleProperty`, not `itemTitleProperty`).